shell/shared/demo.js

```diff
@@ -162,3 +176,34 @@ Router.map(function () {
     }
   });
 });
+
+Router.map(function () {
+  this.route("appdemo", {
+    path: "/appdemo/:packageId",
+    waitOn: function () {
+      return Meteor.subscribe("packageInfo", this.params.packageId);
+    },
+    onAfterAction: function() {
+      /* Stash the packageId into the Meteor session, so that, after
+	 creating a new demo user, we know which app to start a grain
+	 for.
+      */
+      Session.set("packageId", this.params.packageId);
```

Comment:
Hmm, I don't think you need this. Instead, have your data() function include `packageId` in the object it returns. You can access that from your template helper as `this.params.packageId`, I think.

Generally session variables are a way to record actions that the user did which affect the rendering but aren't recorded anywhere else, e.g. "clicked the menu button to expand the menu".